lib/update_util/dfu_file.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include "dfu_headers.h"
+
+#include <stdbool.h>
+#include <storage/storage.h>
+
+typedef enum {
+    UpdateBlockResult_Unknown,
+    UpdateBlockResult_OK,
+    UpdateBlockResult_Skipped,
+    UpdateBlockResult_Failed
+} DfuUpdateBlockResult;
+
+typedef bool (
+    *DfuPageTaskCb)(const uint8_t i_page, const uint8_t* update_block, uint16_t update_block_len);
+typedef void (*DfuPageTaskProgressCb)(const uint8_t progress, void* context);
+typedef bool (*DfuAddressValidationCb)(const size_t address);
+
+typedef struct {
+    DfuPageTaskCb task_cb;
+    DfuPageTaskProgressCb progress_cb;
+    DfuAddressValidationCb address_cb;
+    void* context;
+} DfuUpdateTask;
+
+typedef struct {
+    uint16_t vendor;
+    uint16_t product;
+    uint16_t device;
```

Comment:
`const uint16_t`? Otherwise, it may end up in RAM.

Reply:
It will end up in RAM in any case, that code is for RAM payload.
Compiler enforces all `const` struct members to be immodifiable, so there's no particular reason to make entire struct declaration that strict.

Reply:
![image](https://user-images.githubusercontent.com/277532/162477660-77806501-34b0-4cd9-a372-39d671cf78dd.png)

Reply:
OK